org.eclipse.m2e.core/src/org/eclipse/m2e/core/embedder/IMaven.java

```diff
@@ -148,11 +149,42 @@ MavenExecutionPlan calculateExecutionPlan(MavenProject project, List<String> goa
   MojoExecution setupMojoExecution(MavenProject project, MojoExecution execution, IProgressMonitor monitor)
       throws CoreException;
 
+  /**
+   * Instances should not be cached
+   *
+   * @since 2.1
+   */
+  public interface ConfigurationElement {
+    boolean exists();
+
+    // or firstChild() ? That's what it is actually but it does not look so nice when chained
+    ConfigurationParameter get(String name) throws NoSuchElementException;
+
+    // TODO: or getAll() or similar.
+    Stream<ConfigurationParameter> children(String name) throws NoSuchElementException;
+
+    Stream<ConfigurationParameter> children() throws NoSuchElementException;
+  }
+
+  public interface ConfigurationParameter extends ConfigurationElement {
+    <T> T as(Class<T> clazz) throws NoSuchElementException;
+  }
+
+  /**
+   * @since 2.1
+   */
+  ConfigurationElement getMojoConfiguration(MavenProject project, MojoExecution mojoExecution, IProgressMonitor monitor)
+      throws CoreException;
+
   /**
    * @since 1.4
    */
```

Comment:
deprecated tag also in javadoc with a reference to the method to be used instead.

Reply:
Done.